### PR TITLE
Devteams preparations.

### PR DESCRIPTION
**Pull request by Jaesin**
This PR contains all of changes to `apigee_m10n` and `apige_m10n_ad_credit` from the `8.x-1.x-devtrams` branch without the devteams work included.

### CODE REVIEW

File: src/Entity/Controller/ContextDependentEntityController.php [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+   * @return \Symfony\Component\HttpFoundation\RedirectResponse
+   *   A redirect response.
+   */
+  public function developerRedirect(RouteMatchInterface $route_match): RedirectResponse {
+    // Gets the entity type ID from the route options.
+    $entity_type_id = $route_match->getRouteObject()->getOption('entity_type_id');
+    // The developer routes should always follow this pattern.
+    $developer_route_name = "entity.{$entity_type_id}.developer";
+    // Add the current user as the develoepr (`user`).
+    $parameters = ['user' => \Drupal::currentUser()->id()];
```
Comment (shadcn) [inline src/Entity/Controller/ContextDependentEntityController.php]: Inject this 👍

File: src/Entity/ListBuilder/SubscriptionListBuilderForDeveloper.php [hunk 2]
```diff
@@ -19,309 +19,66 @@
...
    * {@inheritdoc}
    */
-  public static function create(ContainerInterface $container) {
-    $entityType = $container->get('entity_type.manager')->getDefinition('subscription');
-    return static::createInstance($container, $entityType);
-  }
+  public function render(UserInterface $user = NULL) {
+    // Return 404 if the user is not set and keep a compatible method signature.
+    if (!($user instanceof UserInterface)) {
+      return new RedirectResponse(Url::fromRoute('system.404'));
```
Comment (shadcn) [inline src/Entity/ListBuilder/SubscriptionListBuilderForDeveloper.php]: Do we need a redirect here? What about an exception? `throw new NotFoundHttpException();`

Comment (shadcn) [reply src/Entity/ListBuilder/SubscriptionListBuilderForDeveloper.php]: RedirectResponse requires a string: `return new RedirectResponse(Url::fromRoute('system.404')->toString());`

File: src/Entity/ParamConverter/SubscriptionConverter.php [hunk 3]
```diff
@@ -37,39 +34,24 @@ class SubscriptionConverter extends EntityConverter implements ParamConverterInt
...
-    // If the entity type is translatable, ensure we return the proper
-    // translation object for the current context.
-    if ($entity instanceof EntityInterface && $entity instanceof TranslatableInterface) {
-      $entity = $this->entityManager->getTranslationFromContext($entity, NULL, ['operation' => 'entity_upcast']);
-    }
-
-    return $entity;
+    $user = (!$user || $user instanceof UserInterface) ? $user : User::load($user);
+    // Get the developer ID.
+    $develoepr_id = $user instanceof UserInterface ? $user->getEmail() : FALSE;
```
Comment (shadcn) [inline src/Entity/ParamConverter/SubscriptionConverter.php]: s/develoepr_id/developer_id

File: src/Entity/Form/UnsubscribeConfirmForm.php [hunk 4]
```diff
@@ -35,13 +35,6 @@
  */
 class UnsubscribeConfirmForm extends EntityConfirmFormBase {
```
Comment (shadcn) [inline src/Entity/Form/UnsubscribeConfirmForm.php:36]: I'm seeing the following error when I try to unsubscribe:

`Error while cancelling plan: Property "keepOriginalStartDate" has not been initialized on instance of Apigee\Edge\Api\Monetization\Entity\StandardRatePlanRevision class. Expected type: "boolean".`

**Review by shadcn — CHANGES_REQUESTED** [review]
Left some feedback.

File: apigee_m10n.permissions.yml [hunk 5]
```diff
@@ -1,9 +1,6 @@
 administer apigee monetization:
   title: 'Administer apigee monetization'
 
-access monetization packages:
-  title: 'Access monetization packages'
-
 view mint prepaid reports:
   title: 'View Mint Prepaid Reports'
```
Comment (shadcn) [inline apigee_m10n.permissions.yml:5]: We should probably create a ticket and rename this to "view prepaid balance". Not sure why this is using Mint!

Comment (Jaesin) [reply apigee_m10n.permissions.yml:5]: There is a ticket to look at the permissions. [DRUP-555]

File: src/Entity/Form/UnsubscribeConfirmForm.php (hunk 4 above)
Comment (Jaesin) [reply src/Entity/Form/UnsubscribeConfirmForm.php:36]: I see it when trying to subscribe.

Comment (Jaesin) [reply src/Entity/Form/UnsubscribeConfirmForm.php:36]: This will be fixed in a follow up task.

**Review by shadcn — APPROVED** [review]

**Pull request merged**